services/core/java/com/android/server/locksettings/LockSettingsService.java

```diff
@@ -1707,7 +1707,7 @@ public boolean isDuressPassword(LockscreenCredential credential,boolean onlyVali
             Intent intent = new Intent(Intent.ACTION_FACTORY_RESET);
             intent.setPackage("android");
             intent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
-            intent.putExtra(Intent.EXTRA_REASON, PowerManager.SHUTDOWN_THERMAL_STATE);
+            intent.putExtra(Intent.EXTRA_REASON, PowerManager.SHUTDOWN_REASON_UNKNOWN);
```

Comment:
`Intent.ACTION_FACTORY_RESET` is handled by `MasterClearReceiver.java`. It expects `Intent.EXTRA_REASON` to be `String`, not `int`:
https://github.com/GrapheneOS/platform_frameworks_base/blob/SQ3A.220605.009.B1.2022062200/services/core/java/com/android/server/MasterClearReceiver.java#L78

`MasterClearReceiver` calls `RecoverySystem#rebootWipeUserData()`.
Its doc says:
`* @param reason    the reason for the wipe that is visible in the logs`
https://github.com/GrapheneOS/platform_frameworks_base/blob/SQ3A.220605.009.B1.2022062200/core/java/android/os/RecoverySystem.java#L945

Reply:
fixed now, thanks for looking into it.